Set Vcpkg Write-mode Cache intermittently fails with No route to host (management.azure.com:443)

This is the cause of ~4.4% of recent pipeline failures. It has failed ~3 times in the last 14 days.

On `Validate macOS11_release`
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3787351&view=logs&j=6ce62dd9-340c-5b13-8954-b35addbe5c13&t=ffcc7e07-2106-5c75-ba5c-99b26d10a685
```text
2024-05-14T18:06:34.4782700Z Generating script.
2024-05-14T18:06:34.9134270Z Formatted command: . '/Users/runner/work/1/s/eng/scripts/Set-VcpkgWriteModeCache.ps1'
2024-05-14T18:06:35.1352870Z [command]/usr/local/bin/pwsh -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command . '/Users/runner/work/_temp/7689844f-8eb0-4a53-baa7-36e336185fce.ps1'
2024-05-14T18:06:35.1452750Z File saved!
2024-05-14T18:06:36.1993630Z ##[command]Import-Module -Name /usr/local/share/powershell/Modules/Az.Accounts/2.15.0/Az.Accounts.psd1 -Global
2024-05-14T18:06:37.3146610Z ##[command]Clear-AzContext -Scope Process
2024-05-14T18:06:37.6341000Z ##[command]Clear-AzContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
2024-05-14T18:06:37.7459070Z ##[command] Connect-AzAccount -ServicePrincipal -Tenant *** -ApplicationId *** -FederatedToken ***** -Environment AzureCloud -Scope Process
2024-05-14T18:07:54.1522920Z [31;1mConnect-AzAccount: [0m/Users/runner/work/_tasks/AzurePowerShell_72a1931b-effb-4d2e-8fd8-f8472a07cb62/5.239.9/InitializeAz.ps1:113[0m
2024-05-14T18:07:54.1565980Z [31;1m[0m[36;1mLine |[0m
2024-05-14T18:07:54.1618080Z [31;1m[0m[36;1m[36;1m 113 | [0m     $null = [36;1mConnect-AzAccount -ServicePrincipal -Tenant $endpointObje[0m …[0m
2024-05-14T18:07:54.1679490Z [31;1m[0m[36;1m[36;1m[0m[36;1m[0m[36;1m     | [31;1m             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[0m
2024-05-14T18:07:54.1729210Z [31;1m[0m[36;1m[36;1m[0m[36;1m[0m[36;1m[31;1m[31;1m[36;1m     | [31;1mNo route to host (management.azure.com:443)[0m
2024-05-14T18:07:54.4928610Z 
2024-05-14T18:07:54.4998870Z ##[error]PowerShell exited with code '1'.
2024-05-14T18:07:54.5989780Z [command]/usr/local/bin/pwsh -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command . '/Users/runner/work/_tasks/AzurePowerShell_72a1931b-effb-4d2e-8fd8-f8472a07cb62/5.239.9/RemoveAzContext.ps1'
2024-05-14T18:07:56.4299480Z ##[command]Disconnect-AzAccount -Scope CurrentUser -ErrorAction Stop
2024-05-14T18:07:56.7756170Z [33;1mWARNING: You're using Az version 11.2.0. The latest version of Az is 11.6.0. Upgrade your Az modules using the following commands:
2024-05-14T18:07:56.7758300Z   Update-PSResource Az -WhatIf    -- Simulate updating your Az modules.
2024-05-14T18:07:56.7760330Z   Update-PSResource Az            -- Update your Az modules.[0m
2024-05-14T18:07:56.8035880Z ##[command]Disconnect-AzAccount -Scope Process -ErrorAction Stop
2024-05-14T18:07:56.8103900Z ##[command]Clear-AzContext -Scope Process -ErrorAction Stop
2024-05-14T18:07:57.3343180Z 
2024-05-14T18:07:57.3408820Z ##[section]Finishing: Set Vcpkg Write-mode Cache
```

A couple other failures here, when an unrelated PR got merged:
https://github.com/Azure/azure-sdk-for-cpp/pull/5621
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3789836&view=logs&j=3030a67f-a01d-530b-3898-8a79888b0b82&t=e5475e99-6574-50e2-c8a9-adf431572772

cc @danieljurek

--- a/eng/pipelines/templates/steps/vcpkg.yml
+++ b/eng/pipelines/templates/steps/vcpkg.yml
@@ -15,4 +15,4 @@ steps:
         pwsh: true
       # This step is idempotent and can be run multiple times in cases of
       # failure and partial execution.
-      retryCountOnTaskFailure: 5
+      retryCountOnTaskFailure: 3

--- a/eng/pipelines/templates/steps/vcpkg.yml
+++ b/eng/pipelines/templates/steps/vcpkg.yml
@@ -13,3 +13,6 @@ steps:
         ScriptPath: eng/scripts/Set-VcpkgWriteModeCache.ps1
         azurePowerShellVersion: LatestVersion
         pwsh: true
+      # This step is idempotent and can be run multiple times in cases of
+      # failure and partial execution.
+      retryCountOnTaskFailure: 5

